### ISSUE TITLE
Do not warn if unused variable is `_` in situations where an assignment is unavoidable

### ISSUE THREAD

**Issue by Keno** [tonyhffong/Lint.jl#68]
In some situations it's unavoidable to have an assignment, e.g.

```
[a for _ in b]
a, b, _, c = blah()
```

It's a common idiom to name the variable `_`. I think it might make sense to special case the `declared but not used` warning to ignore `_`.
